text/0000-implied-derive.md

```diff
@@ -0,0 +1,425 @@
+   then the deriving system will see this as `#[derive(Copy, Clone)]`
+   and not `#[derive(Copy, Clone, Clone)]`.
+
+2. `Eq` and `PartialOrd` both imply `PartialEq`, and so if the concatenation is
+   used rather than the union, then the compiler will derive `PartialEq` twice
+   which will result in overlapping impls.
+
+### `#[structural_match]`
+
+[RFC 1445]: https://github.com/rust-lang/rfcs/pull/1445
+
+This RFC supercedes [RFC 1445] in that `#[derive(Eq)]`, where transitively
+applied to the transitive closure of all field types of a type, implies
+`#[structural_match]`. In other words, the requirement of transitively having
+`#[derive(PartialEq, Eq)]` is reduced to `#[derive(Eq)]`.
+
+### Interaction with `#[derive_no_bound]` et al.
+
+[RFC 2353]: https://github.com/rust-lang/rfcs/pull/2353
+
+To ensure consistency, `#[derive_no_bound(Trait)]` as described in [RFC 2353]
+should also imply `#[derive_no_bound(Super)]` where `Super` is the
+transitive super-trait closure of `Trait`. Conversely,
+`#[derive_field_bound(Trait)]` will should imply `#[derive_field_bound(Super)]`.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There are two main drawbacks of this proposal.
+
+## Breakage
+
+This RFC will break some code. Specifically, if a user has already manually
+given an `impl` for a trait in the transitive closure of super-traits for a
+particular trait, then an error will be raised. An example of this situation is:
+
+```rust
+#[derive(Copy)]
+struct Foo;
+
+impl Clone for Foo {
+    fn clone(&self) -> Self { Self {} }
+}
+```
+
+which, in the future, would be equivalent to:
+
+```rust
+#[derive(Copy, Clone)]
+struct Foo;
+
+impl Clone for Foo {
+    fn clone(&self) -> Self { Self {} }
+}
+```
+
+and thus resulting in the error:
+
+```rust
+error[E0119]: conflicting implementations of trait `std::clone::Clone` for type `Foo`:
+ --> src/main.rs:2:20
+  |
+2 |     #[derive(Copy, Clone)]
+  |                    ^^^^^ conflicting implementation for `Foo`
+...
+5 |     impl Clone for Foo {
+  |     ------------------ first implementation here
+```
+
+### Mitigating factors
+
+1. We can do this breakage as part of edition 2018.
+
+2. We can give good error messages and help users to migrate with `rustfix`.
+
+3. It is expected that the breakage will be relatively small because situations
+   where `Copy` is derived but `Clone` is implemented is rare.
+   Furthermore, it `Ord` it could be downright risky to derive `Ord` but
+   manually implement `PartialEq`.
+
+## Readability
+
+Arguably, this RFC optimizes for writing ergonomics instead of reading.
+With this RFC implemented, it will be less clear that some traits are derived
+or even implemented from the source code alone since some derived traits are
+only implicitly derived.
+
+### Mitigating factors
+
+However, with the future work outlined in the subsection [rustdoc improvements],
+this drawback can be mitigated.
+
+## Surprising for Haskell developers
```

Comment:
Does this really need to be called out as a drawback?

Reply:
A drawback however insignificant should be mentioned. I always try to write exhaustive RFCs.

Reply:
In that case, should we call out Ruby modules being able to include/extend other modules in the prior art section? There are likely several other languages that deserve consideration here as well

Reply:
Please link the relevant documentation or create a PR and I will certainly include it in the prior art.

Haskell was specifically mentioned because my familiarity with it and because our deriving system is based on Haskell.